src/main/java/org/terracotta/statistics/derived/MinMaxAverage.java

```diff
@@ -17,24 +17,23 @@
 
 import java.util.concurrent.Executor;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAccumulator;
+import java.util.concurrent.atomic.LongAdder;
 
 import org.terracotta.statistics.ValueStatistic;
 import org.terracotta.statistics.observer.ChainedEventObserver;
 import org.terracotta.statistics.util.InThreadExecutor;
 
-import static java.lang.Double.doubleToLongBits;
-import static java.lang.Double.longBitsToDouble;
-
 /**
  *
  * @author cdennis
  */
 public class MinMaxAverage implements ChainedEventObserver {
 
-  private final AtomicLong maximum = new AtomicLong(Long.MIN_VALUE);
-  private final AtomicLong minimum = new AtomicLong(Long.MAX_VALUE);
-  
-  private final AtomicLong summation = new AtomicLong(doubleToLongBits(0.0));
+  private final LongAccumulator maximum = new LongAccumulator(Math::max, Long.MIN_VALUE);
+  private final LongAccumulator minimum = new LongAccumulator(Math::min, Long.MAX_VALUE);
+
+  private final LongAdder summation = new LongAdder();
```

Comment:
This needs to be a DoubleAdder... otherwise we will hit problems at overflow.

Reply:
Done. In fact we should probably add the continuous average instead. That would prevent overflow. But for now, this will do.